puzzle-utils/words/dropquote.py

```diff
@@ -0,0 +1,193 @@
+from .search import search
+
+class DropQuoteSnapshot:
+    def __init__(self):
+        self.grid = []
+        self.letters = []
+        self.words = {}
+    
+    def print(self):
+        print(self.grid)
+        print(self.letters)
+        for index, words in self.words.items():
+            print(f'{index}: {words.letters} | {words.words}')
+        print()
+
+class DropQuoteWords:
+    def __init__(self, letters, words):
+        self.letters = letters
+        self.words = words
+
+def dropquote(grid, letters):
+    """
+    This solver will attempt to solve the drop quote puzzle and returns a set of snapshots on the iterations performed.
+    >>> grid = ['    -', '  -  ', '     ']
```

Comment:
It's hard to see how many spaces there are. This can be a problem when trying to input a grid. Consider using different representations for blanks. Perhaps `0` or `.` (the latter for regex intuitiveness)

Then, consider using a constant for this module to define what is a blank and what is not so we don't have magic strings everywhere